Document Kubeconfig for Automation (#4097)
gravitational/teleport: docs/4.3/cli-docs.md

| `-A, --forward-agent` | none | none | Forward agent to target node like `ssh -A`
| `-L, --forward` | none | none | Forward localhost connections to remote server
| `-D, --dynamic-forward ` | none | none | Forward localhost connections to remote server using SOCKS5
| `-N, -no-remote-exec` | none | none | Don't execute remote command, useful for port forwarding
| `--local` | none |  | Execute command on localhost after connecting to SSH node
| `-t, --tty` | `file` |  | Allocate TTY
| `--cluster` | none |  | Specify the cluster to connect
| `-o, --option` | `local` |  |  OpenSSH options in the format used in the configuration file
| `--enable-escape-sequences` | | | Enable support for SSH escape sequences. Type '~?' during an SSH session to list supported sequences.

> | `--use-local-ssh-agent` | | | Load generated SSH certificates into the local ssh-agent (specified via `$SSH_AUTH_SOCK`). You can also set `TELEPORT_USE_LOCAL_SSH_AGENT` environment variable. Default is true.

Reviewer: ```suggestion
| `--no-use-local-ssh-agent` | | | Do not load generated SSH certificates into the local ssh-agent (specified via `$SSH_AUTH_SOCK`). Useful when using `gpg-agent` or Yubikeys. You can also set the `TELEPORT_USE_LOCAL_SSH_AGENT` environment variable to `false` (default `true`)
```

How about this? `--use-local-ssh-agent` is the default, so I figure we should be explicit about how to disable it rather than enable it.

Author: Yeah, I like this.

> | `--use-local-ssh-agent` | none | true/false | Load generated SSH certificates into the local ssh-agent (specified via `$SSH_AUTH_SOCK`). You can also set `TELEPORT_USE_LOCAL_SSH_AGENT` environment variable. Default is true.

Reviewer: This isn't actually how this flag works and is a bit confusing. The default is `--use-local-ssh-agent` - to disable the functionality, you actually have to pass `--no-use-local-ssh-agent`.

I _believe_ that the environment variable works the way you've written.

@awly Please confirm?

Reviewer 2: Yeah, our flag parsing library doesn't handle `true/false` values for boolean flags, only `--flag/--no-flag`.
And yes, env vars need `true/false` values, opposite to flags.

Author: Thanks for the clarification, I simply removed the default / allowed values.

# Suppress the opening of the system default browser for external provider logins
$ tsh --proxy=proxy.example.com --browser=none

# Login to cluster and output a local kubeconfig

> $ tsh0 login --proxy=proxy.example.com --format=kubernetes -o kubeconfig

Reviewer 3: `tsh0`?

Author: ty, fixed

| Name | Default Value(s) | Allowed Value(s) | Description
|------|---------|----------------|----------------------------|

> | `--k8s-groups` | none | a kubernetes group | Kubernetes groups to assign to a user.e.g. `system:masters`

Reviewer: ```suggestion
| `--k8s-groups` | none | a kubernetes group | Kubernetes groups to assign to a user, e.g. `system:masters`
```

> | `--k8s-users` | none | a kubernetes user | Kubernetes user to assign to a user.e.g. `jenkins`

Reviewer: ```suggestion
| `--k8s-users` | none | a kubernetes user | Kubernetes user to assign to a user, e.g. `jenkins`
```

| Name | Default Value(s) | Allowed Value(s) | Description
|------|---------|----------------|----------------------------|

> | `--type` | none | `trusted_cluster` , `node`   | Type of token to add

Reviewer 3: The `proxy` type should probably be listed here as well.

Author: ty, fixed

> | `--type` | none | `trusted_cluster`, `node`, `proxy`   | Type of token to add

Reviewer: ```suggestion
| `--type` | none | `trusted_cluster`, `node`, `proxy` | Type of token to add
```

Removed additional spaces.

`--user` | none | existing user | Teleport user name
`--host` | none | auth host | Teleport host name
`-o, --out` | none | filepath | identity output

> `--format` | `file` | `file`, `openssh`, `tls` or `kubernetes` |  identity format

Reviewer: ```suggestion
`--format` | `file` | `file`, `openssh`, `tls` or `kubernetes` | identity format
```

# create a certificate with a TTL of 10 years for the jenkins user
# the jenkins.pem file can later be used with `tsh`
$ tctl auth sign --ttl=87600h --user=jenkins --out=jenkins.pem
# create a certificate with a TTL of 1 day for the jenkins user
# the jenkins.pem file can later be used with `tsh`
$ tctl auth sign --ttl=24h --user=jenkins --out=jenkins.pem
# create a certificate with a TTL of 1 day for the jenkins user

> # The kubeconfig file can later be used with kubectl or compatible tooling.

Reviewer: ```suggestion
# The kubeconfig file can later be used with `kubectl` or compatible tooling.
```
